app/validators/domain_block_validator.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class DomainBlockValidator < ActiveModel::Validator
```

Comment:
seems like this could be much more easily handled by adding a reject_media? method that is defined as (media_only? || @reject_media?)